vsp.go

```diff
@@ -440,18 +682,187 @@ func (v *VSP) GetFeeAddress(ctx context.Context, ticketHash chainhash.Hash) (dcr
+	var valueOut valueOut
+
+	// check if host already exists
+	_ = mw.ReadUserConfigValue(VSPHostConfigKey, &valueOut)
+	for _, vsp := range valueOut.List {
+		if vsp == host {
+			return fmt.Errorf("existing host %s", host)
+		}
+	}
+
+	// validate host network
+	info, err := mw.getInfo(host)
+	if err != nil {
+		return err
+	}
+
+	if info.Network != net {
+		return fmt.Errorf("invalid net %s", info.Network)
+	}
+
+	valueOut.List = append(valueOut.List, host)
+	mw.SaveUserConfigValue(VSPHostConfigKey, valueOut)
+	(*mw.VspList).List = append((*mw.VspList).List, &VSPInfo{
+		Host: host,
+		Info: info,
+	})
+
+	return nil
+}
+
+func (mw *MultiWallet) GetRememberVSP() string {
+	var valueOut valueOut
+
+	mw.ReadUserConfigValue(VSPHostConfigKey, &valueOut)
+	return valueOut.Remember
+}
+
+func (mw *MultiWallet) RememberVSP(host string) {
+	var valueOut valueOut
+
+	err := mw.ReadUserConfigValue(VSPHostConfigKey, &valueOut)
+	if err != nil {
+		log.Error(err.Error())
+	}
+
+	valueOut.Remember = host
+	mw.SaveUserConfigValue(VSPHostConfigKey, valueOut)
+}
+
+func (mw *MultiWallet) getInfo(host string) (*VspInfoResponse, error) {
+	sourceWallet := mw.WalletWithID(mw.OpenedWalletIDsRaw()[0])
+	if sourceWallet == nil {
+		return nil, fmt.Errorf("wallet doesn't exist")
+	}
+
+	ctx, cancel := mw.contextWithShutdownCancel()
+	mw.cancelFuncs = append(mw.cancelFuncs, cancel)
+
+	vsp := &VSP{}
+	vsp.vspClient = newVSPClient(host, nil, sourceWallet.Internal())
+	vspInfo, err := vsp.GetInfo(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return vspInfo, nil
+}
+
+func (mw *MultiWallet) SetAutoTicketsBuyerConfig(vspHost string, walletID int, purchaseAccount int32, amountToMaintain int64) {
+	mw.SetLongConfigValueForKey(TicketBuyerATMConfigKey, amountToMaintain)
+	mw.SetIntConfigValueForKey(TicketBuyerWalletConfigKey, walletID)
+	mw.SetInt32ConfigValueForKey(TicketBuyerAccountConfigKey, purchaseAccount)
+	mw.SetStringConfigValueForKey(TicketBuyerVSPHostConfigKey, vspHost)
+	mw.SetBoolConfigValueForKey(TicketBuyerConfigSet, true)
+}
+
+func (mw *MultiWallet) GetAutoTicketsBuyerConfig() (vspHost string, walletID int, purchaseAccount int32, amountToMaintain int64) {
+	atm := mw.ReadLongConfigValueForKey(TicketBuyerATMConfigKey, -1)
+	walId := mw.ReadIntConfigValueForKey(TicketBuyerWalletConfigKey, -1)
+	accNum := mw.ReadInt32ConfigValueForKey(TicketBuyerAccountConfigKey, -1)
+	vspHost = mw.ReadStringConfigValueForKey(TicketBuyerVSPHostConfigKey)
+	return vspHost, walId, accNum, atm
+}
+
+func (mw *MultiWallet) TicketBuyerConfigIsSet() bool {
+	return mw.ReadBoolConfigValueForKey(TicketBuyerConfigSet, false)
+}
+
+func (mw *MultiWallet) ClearTicketBuyerConfig() {
+	mw.SetLongConfigValueForKey(TicketBuyerATMConfigKey, -1)
+	mw.SetIntConfigValueForKey(TicketBuyerWalletConfigKey, -1)
+	mw.SetInt32ConfigValueForKey(TicketBuyerAccountConfigKey, -1)
+	mw.SetStringConfigValueForKey(TicketBuyerVSPHostConfigKey, "")
+	mw.SetBoolConfigValueForKey(TicketBuyerConfigSet, false)
```

Comment:
It seems the only time `TicketBuyerConfigSet` is false is also when `TicketBuyerVSPHost` is cleared. Can do away with the extra config field and just check if the host is empty.